x-pack/plugins/ml/public/application/components/anomaly_results_view_selector/anomaly_results_view_selector.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+import { encode } from 'rison-node';
+
+import { EuiButtonGroup } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { useUrlState } from '../../util/url_state';
+
+interface Props {
+  viewId: 'timeseriesexplorer' | 'explorer';
+}
+
+// Component for rendering a set of buttons for switching between the Anomaly Detection results views.
+
+const toggleButtonsIcons = [
+  {
+    id: 'timeseriesexplorer',
+    label: i18n.translate('xpack.ml.anomalyResultsViewSelector.singleMetricViewerLabel', {
+      defaultMessage: 'View results in the Single Metric Viewer',
+    }),
+    iconType: 'stats',
+    value: 'timeseriesexplorer',
+    'data-test-subj': 'mlAnomalyResultsViewSelectorSingleMetricViewer',
+  },
+  {
+    id: 'explorer',
+    label: i18n.translate('xpack.ml.anomalyResultsViewSelector.singleMetricViewerLabel', {
+      defaultMessage: 'View results in the Anomaly Explorer',
+    }),
+    iconType: 'tableOfContents',
+    value: 'explorer',
+    'data-test-subj': 'mlAnomalyResultsViewSelectorExplorer',
+  },
+];
```

Comment:
it should be defined inside of `AnomalyResultsViewSelector` I guess.
to make sure it is initialized only once use `useMemo`
```typescript
const toggleButtonsIcons = useMemo(() => [
  {
    id: 'timeseriesexplorer',
    label: i18n.translate('xpack.ml.anomalyResultsViewSelector.singleMetricViewerLabel', {
      defaultMessage: 'View results in the Single Metric Viewer',
    }),
    iconType: 'stats',
    value: 'timeseriesexplorer',
    'data-test-subj': 'mlAnomalyResultsViewSelectorSingleMetricViewer',
  },
  {
    id: 'explorer',
    label: i18n.translate('xpack.ml.anomalyResultsViewSelector.singleMetricViewerLabel', {
      defaultMessage: 'View results in the Anomaly Explorer',
    }),
    iconType: 'tableOfContents',
    value: 'explorer',
    'data-test-subj': 'mlAnomalyResultsViewSelectorExplorer',
  },
], []);
```

Reply:
Done inhttps://github.com/elastic/kibana/pull/68663/commits/e67c1fb40fd8180d877148fc18d379610b21e5d6